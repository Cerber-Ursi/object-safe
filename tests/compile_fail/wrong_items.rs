use object_safe::object_safe;

#[object_safe]
fn test() {}

#[object_safe]
struct Test;

#[object_safe]
type TestType = Test;

fn main() {}