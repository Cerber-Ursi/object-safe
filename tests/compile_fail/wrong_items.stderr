error: Only traits can be made object-safe
 --> $DIR/wrong_items.rs:4:1
  |
4 | fn test() {}
  | ^^^^^^^^^^^^

error: Only traits can be made object-safe
 --> $DIR/wrong_items.rs:7:1
  |
7 | struct Test;
  | ^^^^^^^^^^^^

error: Only traits can be made object-safe
  --> $DIR/wrong_items.rs:10:1
   |
10 | type TestType = Test;
   | ^^^^^^^^^^^^^^^^^^^^^
